### PR TITLE
docs: [Documentation changes in RULES.md and NOTICES.md]

### PR DESCRIPTION
**Pull request by isabelle-dr**
**Summary:**

This PR includes minor documentation changes in RULES.md and NOTICES.md that were noticed while analysing the current rules present on the validator. 
This also closes:
- https://github.com/MobilityData/gtfs-validator/issues/916
- https://github.com/MobilityData/gtfs-validator/issues/919
- https://github.com/MobilityData/gtfs-validator/issues/943

**Expected behaviour:** 
Clearer documentation

Please make sure these boxes are checked before submitting your pull request - thanks!

- ~~[ ] Run the unit tests with `gradle test` to make sure you didn't break anything~~
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~~

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -131,7 +131,6 @@ Additional details regarding the notices' context is provided in [`NOTICES.md`](
 | [`RouteShortAndLongNameEqualNotice`](#RouteShortAndLongNameEqualNotice)           	| Short and long name are equal for a route.                                                                                                                  	|
 | [`RouteShortNameTooLongNotice`](#RouteShortNameTooLongNotice)                     	| Short name of a route is too long (more than 12 characters).                                                                                                	|
 | [`SameNameAndDescriptionForStopNotice`](#SameNameAndDescriptionForStopNotice)         | Same name and description for route.                                                                                                                      	|
-| [`StartAndEndTimeEqualNotice`](#StartAndEndTimeEqualNotice)                       	| Equal `frequencies.start_time` and `frequencies.end_time`.                                                                                                  	|
```
Comment (isabelle-dr) [inline RULES.md:134]: This has been depreciated, covered by StartAndEndRangeEqualNotice

File: RULES.md [hunk 2]
```diff
@@ -118,7 +118,7 @@ Additional details regarding the notices' context is provided in [`NOTICES.md`](
...
 | [`DuplicateRouteNameNotice`](#DuplicateRouteNameNotice)                           	| Duplicate  `routes.route_long_name`. Duplicate `routes.route_short_name`. Duplicate combination of fields `route_long_name`  and `routes.route_short_name`. 	|
 | [`EmptyColumnNameNotice`](#EmptyColumnNameNotice)                                 	| A column name is empty.                                                                                                                                     	|
-| [`EmptyRowNotice`](#EmptyRowNotice)                                               	| A file is unknown.                                                                                                                                          	|
+| [`EmptyRowNotice`](#EmptyRowNotice)                                               	| A row in the input file has only spaces.                                                                                                                                          	|
```
Comment (lionel-nj) [inline RULES.md:121]: Definitely ot a big deal but I usually copy-paste the markdown in an external tool to ensure the raw `RULES.md` keeps looking good (with all "|" characters in tables aligned). You can use the following https://www.tablesgenerator.com/markdown_tables to do so.

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Thanks for working on that @isabelle-dr! One small change: could you reformat the tables with the following tool please? https://www.tablesgenerator.com/markdown_tables

File: RULES.md (hunk 2 above)
Comment (isabelle-dr) [reply RULES.md:121]: Interesting! Will do

Comment (isabelle-dr) [reply RULES.md]: It's done, I also fixed the typos in NOTICES.md, as mentioned in https://github.com/MobilityData/gtfs-validator/issues/943

**Comment by isabelle-dr** [comment]
The conflicts are resolved! 🙌
I also moved the SameNameAndDescriptionForRoute from ERROR to WARNING in the section below the tables in RULES.md.

**Review by lionel-nj — APPROVED** [review]
Thanks @isabelle-dr

**Pull request merged**